openupgradelib/openupgrade.py

```diff
@@ -991,17 +1009,46 @@ def wrapped_function(cr, version):
             logger.info(
                 "%s: %s-migration script called with version %s" %
                 (module, stage, version))
+            versioncheck = (version_info[0] >= 8 and v9env) or (
+                version_info > 9 and v9env is not False
+            )
+            if not versioncheck:
+                logger.warning(
+                    "You are are using the decorator without the Environment "
+                    "creation feature. As of Odoo Version 10 Migrations, the "
+                    "Environment will be enabled by default. Make sure to "
+                    "get acquainted with the slightly different migrate() "
+                    "signature in time: `def migrate(env, *args, **kwargs)`"
+                )
             try:
                 # The actual function is called here
                 if hasattr(cr, 'savepoint'):
                     with cr.savepoint():
-                        func(cr, version)
+                        if versioncheck:
+                            with api.Environment.manage():
+                                context['migrate_version'] = version
+                                env = api.Environment(
+                                    cr, SUPERUSER_ID, context
+                                )
+                                return func(env, *args, **kwargs)
+                        else:
+                            return func(cr, version)
                 else:
                     name = uuid.uuid1().hex
                     cr.execute('SAVEPOINT "%s"' % name)
                     try:
-                        func(cr, version)
-                        cr.execute('RELEASE SAVEPOINT "%s"' % name)
+                        if versioncheck:
```

Comment:
On a closer look: If `cr` has no attribute savepoint, it won't have an environment either. So this was basically a no-op.